partner_invoicing_mode_fourteen_days/tests/test_invoicing_fourteen_days.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from freezegun import freeze_time
+
+from odoo.fields import Datetime
+from odoo.tests.common import TransactionCase
+
+from odoo.addons.partner_invoicing_mode.tests.common import CommonPartnerInvoicingMode
+from odoo.addons.queue_job.tests.common import trap_jobs
+
+
+class TestInvoiceModeTenDays(CommonPartnerInvoicingMode, TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner.invoicing_mode = "fourteen_days"
+        cls.partner2.invoicing_mode = "fourteen_days"
+
+    def test_invoice_mode(self):
+        """
+        Try to invoice the concerned sale orders
+        """
+        self._confirm_and_deliver(self.so1)
+        self.assertFalse(self.so1.invoice_ids)
+        with trap_jobs() as traps:
+            self.SaleOrder.generate_fourteen_days_invoices()
+            traps.enqueued_jobs[0].perform()
```

Comment:
You can maybe use `traps.traps.assert_enqueued_job()` in order to check that the code has effectively been processed and that the job is enqueued. In cas another job would have been enqueued in the meanwhile, it is better to call `traps.perfom_enqueued_jobs()` to perform the job I think.